libraries/botbuilder-dialogs/src/prompts/userTokenAccess.ts

```diff
@@ -14,16 +14,12 @@ import {
     UserTokenClient,
 } from 'botframework-connector';
 
-const ExtendedUserTokenProviderT = t
-    .Record({
-        exchangeToken: t.Function,
-        getSignInResource: t.Function,
-        getUserToken: t.Function,
-        signOutUser: t.Function,
-    })
-    .withGuard<ExtendedUserTokenProvider>((val: unknown): val is ExtendedUserTokenProvider => val != null, {
-        name: 'ExtendedUserTokenProvider',
-    });
+const ExtendedUserTokenProviderT = (z.object({
+    exchangeToken: z.function(),
+    getSignInResource: z.function(),
+    getUserToken: z.function(),
+    signOutUser: z.function(),
+}) as any) as z.ZodType<ExtendedUserTokenProvider>; // eslint-disable-line @typescript-eslint/no-explicit-any
```

Comment:
`zod@v1` doesn't have great built-in support for this, this will be something to look into down the line.

Reply:
I stand corrected! `z.custom` was undocumented :)